tests/api/route-services/work-response.test.js

```diff
@@ -1,90 +1,80 @@
 const AWS = require('aws-sdk');
 const AWSMock = require('aws-sdk-mock');
+const ioClient = require('socket.io-client');
+const ioServer = require('socket.io');
 const WorkResponseService = require('../../../src/api/route-services/work-response');
-const { handlePagination } = require('../../../src/utils/handlePagination');
-
-const { cacheSetResponse } = require('../../../src/utils/cache-request');
-
-jest.mock('../../../src/utils/handlePagination', () => ({
-  handlePagination: jest.fn(),
-}));
+const CacheSingleton = require('../../../src/cache');
 
 jest.mock('../../../src/utils/logging');
+jest.mock('../../../src/cache');
 
-jest.mock('../../../src/utils/cache-request', () => ({
-  cacheSetResponse: jest.fn(),
-  cacheGetRequest: jest.fn(),
-}));
-
-let io;
-let emitSpy;
-let toSpy;
+describe('tests for the work-response service', () => {
+  let io;
+  let client;
 
-const setIoMock = () => {
-  emitSpy = jest.fn();
+  const S3_RESULT_ID = '509520fe-d329-437d-8752-b5868ad59425/48637d30-a88d-481e-b8cc-5eedd7e3af1c';
+  const S3_BUCKET_NAME = 'worker-results-staging';
+  const OBJECT_CONTENT = 'my amazing body';
 
-  const mockEmit = {
-    emit: emitSpy,
+  const responseFormat = {
+    request: {
+      uuid: '55',
+      socketId: null,
+      experimentId: '5e959f9c9f4b120771249001',
+      timeout: '2099-01-01T00:00:00Z',
+      body: {
+        name: 'GetEmbedding',
+        type: 'pca',
+      },
+    },
+    response: {
+      error: false,
+      cacheable: true,
+    },
   };
-  toSpy = jest.fn(() => mockEmit);
 
-  io = {
-    to: toSpy,
-  };
-};
-
-const s3Object = {
-  AcceptRanges: 'bytes',
-  LastModified: '2020-05-09T13:51:33.000Z',
-  ContentLength: 223889,
-  ETag: '"1aaac7d75617d2eea36c3d11de3106ec"',
-  ContentEncoding: 'utf-8',
-  ContentType: 'application/json',
-  Metadata: {},
-  Body: 'my amazing body',
-};
-
-AWSMock.setSDKInstance(AWS);
-const getObjectSpy = jest.fn((x) => x);
+  beforeAll(() => {
+    CacheSingleton.createMock({});
+    AWSMock.setSDKInstance(AWS);
+    io = ioServer.listen(3001);
+  });
 
-describe('tests for the work-response service', () => {
-  beforeEach(() => {
-    setIoMock();
+  beforeEach((done) => {
+    client = ioClient.connect('http://localhost:3001', {
+      'reconnection delay': 0,
+      'reopen delay': 0,
+      'force new connection': true,
+      transports: ['websocket'],
+    });
+
+    client.on('connect', () => {
+      responseFormat.request.socketId = client.id;
+      done();
+    });
   });
+
   afterEach(() => {
-    AWSMock.restore('S3');
-    jest.resetModules();
-    jest.resetAllMocks();
-    getObjectSpy.mockReset();
+    AWSMock.restore();
+
+    if (client.connected) {
+      client.disconnect();
+    }
   });
 
   it('Throws during validation if invalid data is supplied', async (done) => {
     try {
-      // eslint-disable-next-line no-unused-vars
-      const w = new WorkResponseService(null, {});
+      // eslint-disable-next-line no-new
```

Comment:
do we still need this eslint-disable there?

Reply:
yes, there is a "don't use constructor for side effects" warning otherwise